src/utils/safeHref.ts

```diff
@@ -0,0 +1,85 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+let useSafeHref = false
+
+export function setUseSafeHref(value: boolean) {
+    useSafeHref = value
+}
+
+export function getUseSafeHref(): boolean {
+    return useSafeHref
+}
+
+export function getURLInfo(url: string): URLInfo {
+    /**
+     * An array of the safely allowed url protocols
+     */
+    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:', 'data:']
+
+    /**
+     * Create an anchor element to parse the protocol and origin of the url provided
+     */
+    const anchorElement = document.createElement('a')
+    anchorElement.href = url
+    const sameOrigin = anchorElement.origin === window.location.origin
```

Comment:
Won't this break server-side rendering support?

Reply:
Good point, added an update for that

Reply:
Let me know if you see any issues with that fix